File: VB/Controllers/HomeController.vb
Imports System.Drawing
Imports System.IO
Imports System.Web.Mvc
Imports CS.Model
Imports DevExpress.Web.Mvc
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrintingLinks

Namespace CS.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View(New MyViewModel With {.Categories = MyModel.GetCategories()})
		End Function
		Public Function GridViewPartialCategories() As ActionResult
			Return PartialView(MyModel.GetCategories())
		End Function

		Public Function ExportTo() As ActionResult
			Dim ps As New PrintingSystem()

			Dim headerImageLink As New Link(ps)

			AddHandler headerImageLink.CreateDetailArea, AddressOf headerImageLink_CreateDetailArea

			Dim link1 As New PrintableComponentLink(ps)
			Dim categoriesGridSettings As New GridViewSettings()
			categoriesGridSettings.Name = "gvCategories"
			categoriesGridSettings.KeyFieldName = "CategoryID"
			categoriesGridSettings.Columns.Add("CategoryID")
			categoriesGridSettings.Columns.Add("CategoryName")
			categoriesGridSettings.Columns.Add("Description")
			link1.Component = GridViewExtension.CreatePrintableObject(categoriesGridSettings, MyModel.GetCategories())

			Dim compositeLink As New CompositeLink(ps)
			compositeLink.Links.AddRange(New Object() { headerImageLink, link1 })
			compositeLink.CreateDocument()

			Dim result As FileStreamResult = CreateExcelExportResult(compositeLink)
			ps.Dispose()

			Return result
		End Function

		Protected Sub headerImageLink_CreateDetailArea(ByVal sender As Object, ByVal e As CreateAreaEventArgs)
			e.Graph.DrawImage(SystemIcons.Application.ToBitmap(), New RectangleF(0, 0, 100, 50))
		End Sub

		Protected Function CreateExcelExportResult(ByVal link As CompositeLink) As FileStreamResult
			Dim stream As New MemoryStream()
			link.PrintingSystem.ExportToXls(stream)
			stream.Position = 0
			Dim result As New FileStreamResult(stream, "application/xls")
			result.FileDownloadName = "MyData.xls"
			Return result
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Drawing;
using System.IO;
using System.Web.Mvc;
using CS.Model;
using DevExpress.Web.Mvc;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrintingLinks;

namespace CS.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View(new MyViewModel { Categories = MyModel.GetCategories() });
        }
        public ActionResult GridViewPartialCategories() {
            return PartialView(MyModel.GetCategories());
        }

        public ActionResult ExportTo() {
            PrintingSystem ps = new PrintingSystem();

            Link headerImageLink = new Link(ps);

            headerImageLink.CreateDetailArea += headerImageLink_CreateDetailArea;

            PrintableComponentLink link1 = new PrintableComponentLink(ps);
            GridViewSettings categoriesGridSettings = new GridViewSettings();
            categoriesGridSettings.Name = "gvCategories";
            categoriesGridSettings.KeyFieldName = "CategoryID";
            categoriesGridSettings.Columns.Add("CategoryID");
            categoriesGridSettings.Columns.Add("CategoryName");
            categoriesGridSettings.Columns.Add("Description");
            link1.Component = GridViewExtension.CreatePrintableObject(categoriesGridSettings, MyModel.GetCategories());

            CompositeLink compositeLink = new CompositeLink(ps);
            compositeLink.Links.AddRange(new object[] { headerImageLink, link1 });
            compositeLink.CreateDocument();

            FileStreamResult result = CreateExcelExportResult(compositeLink);
            ps.Dispose();

            return result;
        }

        protected void headerImageLink_CreateDetailArea(object sender, CreateAreaEventArgs e) {
            e.Graph.DrawImage(SystemIcons.Application.ToBitmap(), new RectangleF(0, 0, 100, 50));
        }

        protected FileStreamResult CreateExcelExportResult(CompositeLink link) {
            MemoryStream stream = new MemoryStream();
            link.PrintingSystem.ExportToXls(stream);
            stream.Position = 0;
            FileStreamResult result = new FileStreamResult(stream, "application/xls");
            result.FileDownloadName = "MyData.xls";
            return result;
        }
    }
}